src/App.tsx

```diff
@@ -1,223 +1,101 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import { getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { NewTodoField } from './components/NewTodoField/NewTodoField';
+import {
+  ErrorNotification,
+} from './components/ErrorNotification/ErrorNotification';
+import { Footer } from './components/Footer/Footer';
+import { TodoInfo } from './components/TodoInfo/TodoInfo';
+// import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [filterBy, setFilterBy] = useState('All');
+  const hasTodos = todos.length > 0;
+  const [mustRenderList, setMustRenderList] = useState<unknown>(null);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
-  }, []);
+
+    if (!user) {
+      return;
+    }
+
+    getTodos(user.id)
+      .then(response => {
+        setTodos(response);
+      })
+      .catch(reject => {
+        setError(`${reject}`);
+        const timeout = setTimeout(() => setError(''), 3000);
+
+        return () => {
+          clearTimeout(timeout);
+          setError('');
+        };
+      });
+  }, [mustRenderList]);
```

Comment:
you use it only in the dependencies array
why not pass `setTodos` and set it directly wherever you need it?
and here just leave empty array